crypto/verify.go

```diff
@@ -3,23 +3,19 @@ package crypto
 import (
 	"github.com/centrifuge/go-centrifuge/crypto/ed25519"
 	"github.com/centrifuge/go-centrifuge/crypto/secp256k1"
+	"github.com/ethereum/go-ethereum/common"
 	"github.com/ethereum/go-ethereum/common/hexutil"
 )
 
 // VerifyMessage verifies message using the public key as per the curve type.
-// if ethereumVerify is true, ethereum specific verification is done
-func VerifyMessage(publicKey, message []byte, signature []byte, curveType string, ethereumVerify bool) bool {
-	signatureBytes := make([]byte, len(signature))
-	copy(signatureBytes, signature)
-
+// for Secp256K1 curve the verification is done with ethereum prefix
+// for Secp256K1, public key should be the address of the original public key, following ethereum standards
+func VerifyMessage(publicKey, message []byte, signature []byte, curveType string) bool {
 	switch curveType {
 	case CurveSecp256K1:
-		if ethereumVerify {
-			address := secp256k1.GetAddress(publicKey)
-			return secp256k1.VerifySignatureWithAddress(address, hexutil.Encode(signatureBytes), message)
-		}
-
-		return secp256k1.VerifySignature(publicKey, message, signatureBytes)
+		signatureBytes := make([]byte, len(signature))
+		copy(signatureBytes, signature)
```

Comment:
this is not needed since you are passing a hex to the Verify, the memory space is different.